src/MapViewController.swift

```diff
@@ -487,6 +495,34 @@ open class MapViewController: UIViewController, LocationManagerDelegate {
     tgViewController.markerRemove(marker)
     return
   }
+
+  //MARK: - private
+
+  private func setupTgControllerView() {
+    tgViewController.view.frame = CGRect(x: 0, y: 0, width: self.view.bounds.width, height: self.view.bounds.height-tabBarHeight)
```

Comment:
Couple issues:
- The reason you're needing to subtract the height here is because you're not using constraints. The view should adopt the size of its superview. So what I would do here is try here is to use the Autresizing Mask to keep it simple. Something like `.autoresizingMask = [.flexibleTopMargin, .flexibleLeftMargin, .flexibleRightMargin, .flexibleBottomMargin]` should do what you want. 

- The next issue is `self.view` doesn't always have a `bounds.width` in viewDidLoad, which is when this is being called. This is because of constraints. You really only get the right size of that once `viewDidLayoutSubviews` is finished. So, if you swap to using constraints, you shouldn't have to set the frame.